channels/jaicp/src/main/kotlin/com/justai/jaicf/channel/jaicp/JaicpConnector.kt

```diff
@@ -31,31 +31,36 @@ abstract class JaicpConnector(
 ) : WithLogger {
 
     private val chatAdapterConnector = ChatAdapterConnector(accessToken, url, httpClient)
-    private val registeredChannels = parseChannels()
+    private var registeredChannels = fetchChannels()
 
-    protected fun registerChannels() {
+    protected fun loadConfig() {
         registeredChannels.forEach { (factory, cfg) ->
-            when (factory) {
-                is JaicpCompatibleChannelFactory -> {
-                    registerChannel(factory.create(botApi), cfg)
-                        .also { logger.info("JAICP-compatible channel has been created for ${factory.channelType}") }
-                }
-
-                is JaicpNativeChannelFactory -> {
-                    registerChannel(factory.create(botApi), cfg)
-                        .also { logger.info("JAICP-native channel has been created for ${factory.channelType}") }
-                }
-
-                is JaicpCompatibleAsyncChannelFactory -> {
-                    registerChannel(factory.create(botApi, getChannelProxyUrl(cfg)), cfg)
-                        .also { logger.info("JAICP-compatible async channel has been created for ${factory.channelType}") }
-                }
-                else -> logger.info("Channel type ${factory.channelType} is not added to list of channels in BotEngine")
-            }
+            createChannel(factory, cfg)
         }
     }
 
-    private fun parseChannels(): List<Pair<JaicpChannelFactory, ChannelConfig>> {
+    private fun createChannel(factory: JaicpChannelFactory, cfg: ChannelConfig) = when (factory) {
+        is JaicpCompatibleChannelFactory -> register(factory.create(botApi), cfg)
+        is JaicpNativeChannelFactory -> register(factory.create(botApi), cfg)
+        is JaicpCompatibleAsyncChannelFactory -> register(factory.create(botApi, getChannelProxyUrl(cfg)), cfg)
+        else -> logger.info("Channel type ${factory.channelType} is not added to list of channels in BotEngine")
+    }
+
+    protected fun reloadConfig() {
+        val fetched = fetchChannels()
+
+        registeredChannels.filter { (_, regCfg) ->
```

Comment:
These lines are pretty hard to read, let's better explain what are we doing here:
```
val stillRegistered = registeredChannels.map { it.second.channel }.intersect(fetched.map { it.second.channel })

registeredChannels.filter { it.second.channel !in stillRegistered }.forEach { 
  evict(it.second) 
}
fetched.filter { it.second.channel !in stillRegistered }.forEach { 
  createChannel(it.first, it.second) 
}

registeredChannels = fetched
```